src/compiler/compile/css/Selector.ts

```diff
@@ -126,47 +126,20 @@ export default class Selector {
 	}
 }
 
-function apply_selector(stylesheet: Stylesheet, blocks: Block[], node: CssNode, stack: CssNode[], to_encapsulate: any[]): boolean {
+function apply_selector(blocks: Block[], node: CssNode, stack: CssNode[], to_encapsulate: any[]): boolean {
 	const block = blocks.pop();
 	if (!block) return false;
 
 	if (!node) {
 		return blocks.every(block => block.global);
 	}
 
-	let i = block.selectors.length;
-
-	while (i--) {
-		const selector = block.selectors[i];
-		const name = typeof selector.name === 'string' && selector.name.replace(/\\(.)/g, '$1');
-
-		if (selector.type === 'PseudoClassSelector' && name === 'global') {
-			// TODO shouldn't see this here... maybe we should enforce that :global(...)
-			// cannot be sandwiched between non-global selectors?
+	try {
```

Comment:
I don't think we want this `try` here. If there's some CSS or markup that causes the TypeError below to be thrown, we'd like to know about it, rather than masking it, because that's a bug. What do you think about that?

Other than that, this looks like a reasonable fix - thank you very much!

Reply:
Oh whoops, sorry, I see. There was no `throw` in the original code.

I still think that catching _all_ `TypeError`s is probably a bit dangerous, and would make other bugs harder to find. I'd prefer if this `catch` keyed into something specific about the error you're throwing below, and would otherwise re-throw the error. Or, perhaps even better, use a third return value from the function besides `true` and `false`.

Reply:
I agree, I didn't love the try/catch/error solution but it was a convenient way to have essentially three return values.

What if we used a kind of enum with eg. `NOT_POSSIBLE`, `POSSIBLE` and `UNKNOWN_SELECTOR_TYPE` or something? This might be nice because I was thinking later we could differentiate between `POSSIBLE` and `DEFINITELY` or something, as a way to optimize this further.

Another option could be to introduce a new error eg. `UnknownSelectorTypeError` to stick with the `try`/`catch` approach but avoid the problems with catching all `TypeError`s. Let me know what you prefer.

Reply:
👍 I like the idea of using an enum.